File: Excel/461 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/461 Sort the Numbers.xlsx", range = "A1:A10")
test  = read_excel("Excel/461 Sort the Numbers.xlsx", range = "B1:B10")

result = input %>%
  separate(String, into = c("A", "B", "C", "D"), sep = "\\.", remove = FALSE) %>%
  mutate(across(A:D, as.numeric)) %>%
  arrange(A, B, C, D) %>%
  select(String)

identical(result$String, test$`Answer Expected`)
# [1] TRUE

File: Excel/400-499/461/461 Challenge.py
import pandas as pd

input = pd.read_excel("461 Sort the Numbers.xlsx", usecols="A", nrows = 9)
test = pd.read_excel("461 Sort the Numbers.xlsx", usecols="B", nrows = 9)

input[["A","B","C","D"]] = input["String"].str.split(".", expand=True).apply(pd.to_numeric)
input = input.sort_values(by=["A","B","C","D"])
result = input["String"].reset_index(drop=True)

print(result.equals(test["Answer Expected"])) # True